docs/content/stable/architecture/transactions/concurrency-control.md

```diff
@@ -1272,6 +1272,76 @@ The `NOWAIT` clause for row-level explicit locking doesn't apply to the `Fail-on
+
+To reduce the overhead of DMLs, YugabyteDB takes shared locks on the PostgreSQL backend's host TServer alone. Global locks are propagated to the Master leader.
+
+Each TServer maintains an in-memory `TSLocalLockManager` that serves object lock acquire or release calls. The Master also maintains an in-memory lock manager primarily for serializing conflicting global object locks. When the Master leader receives a global lock request, it first acquires the lock locally, then fans out the lock request to all TServers with valid [YSQL leases](#ysql-lease-mechanism), and executes the client callback after the lock has been successfully acquired on all TServers (with a valid YSQL lease).
+
+All statements inherently acquire some object locks. You can also explicitly acquire object locks using the [LOCK TABLE API](https://www.postgresql.org/docs/current/sql-lock.html):
+
+```sql
+-- Acquire a share lock on a table
+LOCK TABLE my_table IN SHARE MODE;
+
+-- Acquire an access exclusive lock (blocks all other operations)
+LOCK TABLE my_table IN ACCESS EXCLUSIVE MODE;
+```
+
+You can observe active object locks using the [pg_locks](../../../explore/observability/pg-locks/) system view for transactions older than [yb_locks_min_txn_age](../../../explore/observability/pg-locks/#yb-locks-min-txn-age):
+
+```sql
+SELECT * FROM pg_locks WHERE NOT granted;
+```
+
+### Lock scope and lifecycle
+
+All object locks are tied to a DocDB transaction:
+
+- **DMLs**: Locks are released at commit or abort time
+- **DDLs**: Lock release is delegated to the Master, which has a background task for observing DDL commits or aborts and finalizing schema changes.
+
+When a DDL finishes, all locks corresponding to the transaction are released and this release path enforces cache refresh on the the TServers ensuring that they have the latest catalog cache. This also ensures any new DMLs waiting on the same locks to see the latest schema after acquiring the object locks.
+
+To reduce overhead for read-only workloads, YugabyteDB reuses DocDB transactions wherever possible.
+
+### Failure handling
+
+Locks are cleaned up for various failure scenarios as follows:
+
+- PostgreSQL backend crash: The TServer-PostgreSQL session heartbeat cleans up corresponding locks.
+- TServer crash: YSQL Lease mechanism cleans up DDL locks that originated from the TServer.
+- Master leader crash: Newly elected Master leader replays catalog entries to recreate locks' state held (at the master) and establish the lease for the TServers.
+
+### Important considerations
+
+- If a TServer doesn't have a valid YSQL lease, it cannot serve any requests from PostgreSQL backends.
+- Upon TServer lease changes (due to network issues), all active PostgreSQL backends are killed.
+- If TServers cannot communicate with the Master leader for shorter durations (lesser than YSQL lease), DDLs will stall because global locks cannot be served.
```

Comment:
```suggestion
- If TServers cannot communicate with the Master leader for shorter durations (less than the YSQL lease), DDLs will stall because global locks cannot be served.
```